### PR TITLE
Feature/add pagination to participants list

### PR DESCRIPTION
**Pull request by Morsey187**
[213](https://projects.torchbox.com/projects/flare-rebuild/tickets/213)

#### When applied this MR will...

Add pagination to the participants list page

#### Please provide detail on the technical changes, if relevant

new `DEFAULT_PER_PAGE` env variable. other changes are described in comments.

#### Screenshots

Screenshot 2021-11-11 at 16 53 05
Screenshot 2021-11-11 at 16 52 45

#### Dev checklist

- [x] PR addresses all ACs
~- [ ] Added unit tests if necessary~
~- [ ] Updated documentation if necessary~
- [x] CI passes
- [x] Code reviewed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ Starting a local build can be done by running:
 git clone [URL TO GIT REMOTE]
 cd flare_portal
 fab build
-fab up
```
Comment (Morsey187) [inline README.md:57]: Thought I'd sneak this one in here

File: flare_portal/experiments/views.py [hunk 2]
```diff
@@ -522,17 +524,33 @@ def get_context_data(self, **kwargs: Any) -> dict:
...
+        participants_qs = Participant.objects.filter(
+            experiment=self.experiment,
+        )
+        paginator = Paginator(participants_qs, self.paginate_by)
+        try:
+            participants = paginator.page(self.page)
+        except PageNotAnInteger:
+            participants = paginator.page(1)
+        except EmptyPage:
+            participants = paginator.page(paginator.num_pages)
```
Comment (alxbridge) [inline flare_portal/experiments/views.py:544]: **suggestion:** Handle `InvalidPage` here, to ensure requests for pages out of range don't cause an error.

**Comment by alxbridge** [comment]
Hi @Morsey187 - your code all looks correct to my eye: just a couple of comments relating to some potential improvements.

**Comment by Morsey187** [comment]
> Hi @Morsey187 - your code all looks correct to my eye: just a couple of comments relating to some potential improvements.

Thanks for this @alxbridge , currently pushed to staging to have it tested, will make that change for `InvalidPage` to and hopefully get this pushed tomorrow.

Although regarding `InvalidPage` from a quick scan of the docs and testing it seems like I'm handling this already https://docs.djangoproject.com/en/3.2/ref/paginator/#exceptions and shouldn't be needed

**Comment by alxbridge** [comment]
> Although regarding `InvalidPage` from a quick scan of the docs and testing it seems like I'm handling this already https://docs.djangoproject.com/en/3.2/ref/paginator/#exceptions and shouldn't be needed

Ah yep, looks you're right - I missed the detail that the exception will always be one of the two subclasses you're already handling.

**Review by RealOrangeOne — APPROVED** [review]

**Pull request merged**